src/hooks/useGlobalState.tsx

```diff
@@ -11,15 +15,34 @@ export function withGlobalContext<P>(
 ): (props: P) => JSX.Element {
   return function WrappedComponentWithGlobalState(props: P): JSX.Element {
     const [state, dispatch] = useReducer(reducer, initialStateFunc())
+    // hold a reference to state here
+    const stateRef = useRef(state)
+    // deep check there are no majour changes
+    // between updates
+    const deepState = useDeepCompareRef(state)
+    // set new ref current value if no deep changes
+    useMemo(() => {
+      stateRef.current = deepState as GlobalState
```

Comment:
Why even do it? `deepCompare` can get expensive depending on depth
If it's only to ensure `getState` inside middlewares is current, just do
```js
const stateRef = useRef(state)
stateRef.current = deepState
```

without extra hooks and checking